### ISSUE TITLE
Podman run `--group-entry` substitution `$GROUPNAME` and  `$USERLIST` values empty

### ISSUE THREAD

**Issue by cevich** [containers/podman#18902]
### Issue Description

From the podman-run man page:

```
   --group-entry=ENTRY
       Customize  the  entry that is written to the /etc/group file within the con‐
       tainer when --user is used.

       The variables $GROUPNAME, $GID, and  $USERLIST  are  automatically  replaced
       with their value at runtime if present.
```

However, the `$GROUPNAME` and  `$USERLIST` substitution values are always empty inside a container.  Despite my user (on the host) having access to `/etc/group`, and both a group entry and its user-list field are non-empty (the `id` and `getent group <name>` commands work perfectly fine).

### Steps to reproduce the issue

Steps to reproduce the issue
1. `id`
2. `podman run -it --rm --user $(id -u):$(id -g) --group-entry '$GROUPNAME:x:$GID:$USERLIST' registry.fedoraproject.org/fedora:latest` ***N/B: single-quotes to protect values from shell***
3. (inside container) `id`
4. (inside container) `tail -1 /etc/group`

### Describe the results you received

1. `uid=1234(cevich) gid=1234(cevich) groups=1234(cevich),10(wheel)`
2. (bash prompt)
3. `uid=1234(cevich) gid=1234() groups=1234()`
4. `:x:4179:`

### Describe the results you expected

3. `uid=1234(cevich) gid=1234(cevich) groups=1234(cevich)`
4. `cevich:x:1234:cevich`

### podman info output

```yaml
version:
  APIVersion: 4.5.0
  Built: 1681486976
  BuiltTime: Fri Apr 14 11:42:56 2023
  GitCommit: ""
  GoVersion: go1.19.7
  Os: linux
  OsArch: linux/amd64
  Version: 4.5.0
```
```


### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

No

### Additional environment details

Fedora 37 host

### Additional information

Running the container instead with `--group-entry 'cevich:x:1234:cevich'` works perfectly:

```
bash-5.2$ tail -1 /etc/group
cevich:x:1234:cevich
```

The magic `$GID` token seems to be substituted just fine.  It's only `$GROUPNAME` and  `$USERLIST` that don't seem to be looked up on the host and/or fail to get passed into the container and set in `/etc/group`.

**Comment by Luap99** [comment]
Looking at the code this fills in values from the container not the hosts /etc/group.

**Comment by Luap99** [comment]
Sorry I really meant this in we read /etc/group from within the container to populate these vars. Although I really do not understand how they are supposed to be used. The whole --group-entry flag seems to be a NOP AFAICT.

**Comment by cevich** [comment]
> Sorry I really meant this in we read /etc/group from within the container to populate these vars. Although I really do not understand how they are supposed to be used. 

Yeah, I don't see much utility in the substitution tokens as well.  Presumably if someone is building a special image anyway, they'd simply encode all the necessary groups/users at build time.  OTOH, if they pulled data from the host (as the man page isn't clear about), I could actually see some useful utility in that.

> The whole --group-entry flag seems to be a NOP AFAICT.

Well it's not completely useless, it's just not well documented and/or operates counter-intuitively.  I'd make a similar argument for the `--user <UID>:<GID>` form, but that's largely discussed in https://github.com/containers/podman/issues/18903.

If I had to guess, given the inconsistent behaviors/features in both `--group-entry` and `--user`, it almost feels like we've been fighting with ourselves over docker compatibility vs unique features...so basically failing at both :disappointed:

Anyway, I'm okay with fixing up the docs. to resolve this issue...but it doesn't give me good feelings overall.  Perhaps we should bring the `--group-entry` and `--user` oddities to a community cabal and see if we can arrive at a better direction to take them, maybe for 5.0?

**Comment by danishprakash** [comment]
I agree that the documentation is lacking in many regards for `--group-entry` but while I was adding it, the idea was to simply replace the said variables with the user that's running podman if rootless[1], somewhat similar to how we had implemented `--user`.

@cevich if you try `--user` and `--group-entry` in conjunction, does that still give you the same undesirable output?

**Comment by danishprakash** [comment]
I agree that the documentation is lacking in many regards for `--group-entry` but while I was adding it, the idea was to simply replace the said variables with the user that's running podman if rootless[1], somewhat similar to how we had implemented `--user`.

@cevich if you try `--user` and `--group-entry` in conjunction, does that still give you the same undesirable output? Either way, I'll take a look at this tomorrow.

**Comment by cevich** [comment]
> the idea was to simply replace the said variables with the user that's running podman if rootless[1], somewhat similar to how we had implemented --user.

Ahh okay.  That doesn't appear to be what's happening, it's taking those substitution values from w/in the container (i.e. `ENV` or `--env` values presumably) [according to Paul's reply above.](https://github.com/containers/podman/issues/18902#issuecomment-1593401780)

> Either way, I'll take a look at this tomorrow.

Thanks.  As I mentioned above, a fix would be really cool, if not some updated documentation would also be good (I can help if needed).

**Comment by rhatdan** [comment]
@danishprakash Are you working on a fix for this?
